python/tvm/driver/tvmc/frontends.py

```diff
@@ -161,14 +161,11 @@ def load(self, path):
         # pylint: disable=E1101
         model = onnx.load(path)
 
+        shape_dict = {}
         # pylint: disable=E1101
-        name = model.graph.input[0].name
-
-        # pylint: disable=E1101
-        proto_shape = model.graph.input[0].type.tensor_type.shape.dim
-        shape = [d.dim_value for d in proto_shape]
-
-        shape_dict = {name: shape}
+        for i in model.graph.input:
+            # pylint: disable=E1101
+            shape_dict[i.name] = [dim.dim_value for dim in i.type.tensor_type.shape.dim]
```

Comment:
I wonder if we could make `from_onnx`[1] better, to come up with the shape, we case we pass no shape.

https://github.com/apache/tvm/blob/7a0d10ce8d4df27734949b944992e5c516f620d0/python/tvm/relay/frontend/onnx.py#L2738

Reply:
it is quite naturally not to require shape explicitly if it is already defined in original model and user do not want to redefine it.

There might be a solution to move these three lines into the from_onnx() and verify if shape param is passed or not. if it is not - add logic from the PR.

Reply:
I agree with @leandron. It would be better to keep TVMC as a command line argument processor and put other logic to other modules as we can. The logic @dlexplorer proposed looks good to me. In addition, when the shape param is passed, we could also check if it is consistent with the one defined in the model, or throw out an error or a warning otherwise. In this case, we don't even need a test case in this PR.

Also cc @mbrookhart who recently works on the ONNX frontend.

Reply:
Should we also expose shape as an argument of TVMC，so that we can specify shape from command line?

Reply:
I'm not sure why we need to do that. Could you provide a use case that we need to ask users to specify shape from command line even if the model file already has the information?

Reply:
Some models are defined with dynamic input, like this one https://github.com/onnx/models/tree/master/vision/object_detection_segmentation/yolov3
But we just want to apply it to some specific input sizes, to fix input size, we can reduce dynamism because we can get better performance.

Some frontends like from_onnx and from_tensorflow can accept "shape" as input but we can't make use of that from TVMC

Reply:
I see. It would be better to file another PR to add new features in TVMC.
cc @leandron

Reply:
My understanding is that we all agree two things here:
1) moving the current logic to to `from_onnx`, to be done in the current PR (@dlexplorer )
2) a new feature to include `--input-shape` (in `tvmc compile` and `tvmc tune` I think), to be submitted in a separate PR @lixiaoquan)

Reply:
I understood the same. Sorry, little bit busy these days, cannot modify PR quickly. Hope will do later today

Reply:
@leandron Yes, it is good we all agree to add this option

Reply:
just uploaded a change, will appreciate if you take a look